src/components/NewMovie/NewMovie.tsx

```diff
@@ -1,37 +1,141 @@
-import { useState } from 'react';
+import React, { useState } from 'react';
+import { Movie } from '../../types/Movie';
+import { MovieError } from '../../types/MovieError';
 import { TextField } from '../TextField';
 
-export const NewMovie = () => {
+type Props = {
+  onAdd: (movie: Movie) => void;
+};
+
+const defaultMovie: Movie = {
+  title: '',
+  description: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+const defaultMovieError: MovieError = {
+  title: '',
+  imgUrl: '',
+  imdbUrl: '',
+  imdbId: '',
+};
+
+export const urlRegex =
+  // eslint-disable-next-line max-len
+  /^((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=+$,\w]+@)?[A-Za-z0-9.-]+|(?:www\.|[-;:&=+$,\w]+@)[A-Za-z0-9.-]+)((?:\/[+~%/.\w-_]*)?\??(?:[-+=&;%@,.\w_]*)#?(?:[,.!/\\\w]*))?)$/;
+
+export const NewMovie: React.FC<Props> = ({ onAdd }) => {
   // Increase the count after successful form submission
   // to reset touched status of all the `Field`s
-  const [count] = useState(0);
+  const [count, setCount] = useState(0);
+  const [movie, setMovie] = useState(defaultMovie);
+  const [movieError, setMovieError] = useState(defaultMovieError);
+
+  const hasFormErrors =
+    !movie.title ||
+    !movie.imgUrl ||
+    !movie.imdbUrl ||
+    !movie.imdbId ||
+    !urlRegex.test(movie.imgUrl) ||
+    !urlRegex.test(movie.imdbUrl);
+
+  const setMovieErrorFn = (field: keyof MovieError, error: string) => {
+    setMovieError(prev => {
+      const newMovieError = { ...prev };
+
+      newMovieError[field] = error;
+
+      return newMovieError;
+    });
+  };
+
+  const handleOnChange = (field: keyof Movie, newValue: string): void => {
```

Comment:
and add `type` to `e`
```suggestion
  const handleOnChange = (e): void => {
```